shivyc/il_cmds/math.py

```diff
@@ -118,6 +118,76 @@ class Mult(_AddMult):
     Inst = asm_cmds.Imul
 
 
+class _BitShiftCmd(ILCommand):
+    """Base class for bitwise shift commands."""
+
+    # The ASM instruction to generate for this command. Override this value
+    # in subclasses.
+    Inst = None
+
+    def __init__(self, output, arg1, arg2): # noqa D102
+        self.output = output
+        self.arg1 = arg1
+        self.arg2 = arg2
+
+    def inputs(self): # noqa D102
+        return [self.arg1, self.arg2]
+
+    def outputs(self): # noqa D102
+        return [self.output]
+
+    def rel_spot_pref(self): # noqa D102
+        return {self.output: [self.arg1, self.arg2]}
+
+    def make_asm(self, spotmap, home_spots, get_reg, asm_code): # noqa D102
+        arg1_spot = spotmap[self.arg1]
+        arg1_size = self.arg1.ctype.size
+        arg2_spot = spotmap[self.arg2]
+        arg2_size = self.arg2.ctype.size
+
+        # According Intel® 64 and IA-32 software developer's manual
+        # Vol. 2B 4-582 second (count) operand must be represented as
+        # imm8 or CL register.
+        if not self._is_imm8(arg2_spot) and arg2_spot != spots.RCX:
+            asm_code.add(asm_cmds.Mov(spots.RCX, arg2_spot, arg2_size))
+            arg2_spot = spots.RCX
+
+        # Destination operand must be register or a memory spot.
+        if not (isinstance(arg1_spot, spots.MemSpot) or
```

Comment:
I don't understand why this condition is necessary. If the arg1_spot is not a memory or register spot, then won't it automatically be moved into a memory or register spot by the lines that follow?